Logger/BaseLoggerMixins.cs

```diff
@@ -1,7 +1,57 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
+
     public static class BaseLoggerMixins
     {
-        
+
+        public static void Error(this BaseLogger logNum, string message, params object[] paramArray)
```

Comment:
```suggestion
        public static void Error(this BaseLogger? logger, string message, params object[] paramArray)
```
`logNum` is an odd name. Try to avoid using abbreviated variable names.